### PR TITLE
feat: #251 send webhook notification if event creation failed

### PR DESCRIPTION
**Pull request by lovrocolic**

### CODE REVIEW

File: app/services/events_service.rb [hunk 1]
```diff
@@ -48,4 +64,19 @@ def current_customer(organization_id = nil, customer_id = nil)
...
+    valid_codes.include? code
+  end
+
+  def send_webhook_notice(organization, params)
+    object = {
+      input_params: params,
+      error: result.error
+    }
+
+    SendWebhookJob.perform_later(:event, object, organization) if organization.webhook_url?
```
Comment (lovrocolic) [inline app/services/events_service.rb:80]: activeJob does not accept OpenStruct (our result) so I had to send hash here

**Review by vincent-pochet — COMMENTED** [review]
Looks good! I've just made a suggestion regarding the organization optional parameter

File: app/services/events_service.rb [hunk 2]
```diff
@@ -48,4 +64,19 @@ def current_customer(organization_id = nil, customer_id = nil)
...
+  def valid_code?(code, organization)
+    valid_codes = organization.billable_metrics.all.pluck(:code)
+
+    valid_codes.include? code
+  end
+
+  def send_webhook_notice(organization, params)
+    object = {
+      input_params: params,
+      error: result.error
```
[the remark this answers is not in the document]
Comment (lovrocolic) [reply app/services/events_service.rb]: Tnx, you are right, that's cleaner :)

File: app/services/events_service.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (lovrocolic) [reply app/services/events_service.rb]: hmm, it was already there, did I miss something?

[the remark this answers is not in the document]
Comment (lovrocolic) [reply app/services/events_service.rb]: I see 👍 Updated

File: app/services/webhooks/event_service.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    private
+
+    def current_organization
+      @current_organization ||= Organization.find(object[:organization_id])
+    end
+
+    def object_serializer
+      ::ErrorSerializer.new(
+        OpenStruct.new(object),
+        root_name: 'error_event',
```
Comment (vincent-pochet) [inline app/services/webhooks/event_service.rb:14]: `object_type` and `root_name` should have the same value.

The idea behind this is that the webhook consumer should be able to know easily what attribute to look for in the payload:
```json
{
  "object_type": "event_error",
  "event_error": {
   ...
  }
```

**Review by vincent-pochet — APPROVED** [review]
One last comment. Great job! :rocket:

File: app/services/webhooks/event_service.rb (hunk 3 above)
Comment (lovrocolic) [reply app/services/webhooks/event_service.rb]: Thanks for the clarification, wasn't aware of it :)

**Review by vincent-pochet — APPROVED** [review]

**Review by jdenquin — APPROVED** [review]
all good for me, nice one!

**Pull request merged**